app/workers/segment_delete_user_worker.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+class SegmentDeleteUserWorker < ActiveJob::Base # rubocop:disable Rails/ApplicationJob # That would be Rails 5 :)
+  class MissingUserIDError < StandardError; end
+
+  def perform(event_id)
+    event = EventStore::Repository.find_event!(event_id)
+    raise(MissingUserIDError) unless (user_id = event.data[:user_id])
+    SegmentDeleteService.new(user_id).call!
+  rescue ActiveRecord::RecordNotFound, MissingUserIDError => exception
+    System::ErrorReporting.report_error(exception, parameters: {event_id: event_id})
+  rescue SegmentDeleteService::SegmentDeleteServiceError => exception
```

Comment:
There is no need for this error, well at least it is too generic. Probably we should do:

- Network error
  * Server error => retry the job because this is something we cannot manage
  * Client error => we did something bad, log the error and do not retry the job

- Other type of error: just log and do not retry